msteams-platform/samples/virtual-tables-api.md

```diff
@@ -0,0 +1,229 @@
+
+Examples:
+
+* GET [Organization URI]/api/data/v9.2/m365_graphbookingappointments?$filter=m365_bookingbusinessid eq ‘ContosoBank@Contoso.onmicrosoft.com’ and m365_price eq 100.0
+* GET [Organization URI]/api/data/v9.2/m365_graphdriveitems?$filter=m365_collaborationrootid eq ‘00000000-0000-0000-0000-000000000000’ and m365_name eq ‘Meeting Notes.docx’
+* GET [Organization URI]/api/data/v9.2/m365_graphevents?$filter=m365_groupid eq ‘00000000-0000-0000-0000-000000000000’ and m365_subject eq ‘Monthly Sync’
+
+## Virtual table pagination
+
+Pagination is a useful resource for fetching a large set of records. Virtual Table pagination can be achieved in three different ways.
+
+You can specify the page size by using the `odata.maxpagesize` preference value in the request header. If the result set spans multiple pages, the response will include the `@odata.nextLink` property. Sample request and response are as following:
+
+# [Request](#tab/request)
+
+```http
+  GET [Organization URI]/api/data/v9.2/m365_driveitems 
+  Accept: application/json 
+  Prefer: odata.maxpagesize=2 
+```
+
+# [Response](#tab/response)
+
+```json
+{ 
+
+  "@odata.context": "[Organization URI]/api/data/v9.0/$metadata#m365_graphdriveitems", 
+  "value": [ 
+   { 
+      "@odata.etag": "W/\"{FA93AF7C-1F45-4714-85A5-BB95EB86E1E5}\"", 
+      "m365_name": "Review.doc", 
+      "m365_graphdriveitemid": "f50aae23-6644-3d35-66d7-e3c5a979dad3", 
+      …
+      },
+      {
+      "@odata.etag": "W/\"{3938D549-1AEF-46A5-BF3C-38472AD934C2}\"", 
+      "m365_name": "Review.doc", 
+      "m365_graphdriveitemid": "3d59a7e2-ec83-d0b3-270e-8ad676622027", 
+      … 
+      } 
+      ],
+      "@odata.nextLink": "[Organization URI]/api/data/v9.0/m365_graphdriveitems &$skiptoken=%3Ccookie%20pagenumber=%222%22%20pagingcookie=%22UGFnZWQ9VFJVRSZwX1NvcnRCZWhhdmlvcj0xJnBfRmlsZUxlYWZSZWY9dGVzdCZwX0lEPTI5%22%20istracking=%22False%22%20/%3E" 
+} 
+```
+
+---
+
+Currently the following Virtual Tables support the `odata.maxpagesize` preference:
+
+* Graph Booking Appointment
+* Graph Calendar Event
+* Graph Chat
+* Graph Chat Message
+* Graph Drive
+* Graph Drive Item
+
+You can specify the number of records to return by passing the `$top` option in the URL. If you also need to specify the page number, you can do so by passing a paging cookie as an XML-encoded string as the `$skiptoken` option. To fetch a specific page number, you can pass the paging cookie in the following format:
+
+  `<cookie pagenumber=3 />`
+
+# [Request](#tab/request1)
+
+```http
+     GET [Organization URL]/api/data/v9.2/m365_graphevents?$top=2&$skiptoken=<cookie pagenumber=3> 
+```
+
+# [Response](#tab/response1)
+
+```json
+
+{
+  "@odata.context": "[Organization URI]/api/data/v9.0/$metadata#m365_graphdevents", 
```

Comment:
Small typo - the address should be `[Organization URI]/api/data/v9.0/$metadata#m365_graphevents`